### ISSUE TITLE
Configuring now always takes an insane amount of memory

### ISSUE THREAD

**Issue by leftaroundabout** [haskell/cabal#4589]
After installing the HEAD version of Cabal (8315dbf), it has become practically unusable because any `configure` (or `new-configure`) allocates about 11 GiB of memory, which is quite enough to drive my laptop hopelessly into swap.

`cabal-install` apparently does this _before even doing any of the actual configuration work_, because when there's a syntax error in the `.cabal` file, I only get an error after the memory load is down again.

I thought it might have something to do with old cruft in my `~/.cabal`, but even completely removing that and setting up from scratch didn't change anything. Nor did it help to remove everything in `/tmp`.

System info: Ubuntu 16.04, `ghc-8.3.20170622`.

**Comment by 23Skidoo** [comment]
This doesn't affect 2.0, right?

**Comment by 23Skidoo** [comment]
I wonder why Travis didn't catch this.

**Comment by 23Skidoo** [comment]
@leftaroundabout Does this happen with all packages or only with your project? Can you post repro instructions?

**Comment by leftaroundabout** [comment]
It happens with all packages, (even a `module Main where main = return ()` dummy).

The problem is the same with a 2.0.0.0 version I've just built (with that same HEAD compiler), so it a lot like it has something to do with either GHC-HEAD or my local configuration. Will try compiling `cabal-install` with another compiler now...

**Comment by leftaroundabout** [comment]
Ok, when cabal-install is built with `ghc-8.0.1`, the problem is gone.
```bash
[21:24:19]sagemuej@sagemuej-X302LA:/tmp/cbt  # a basically empty dummy package
$ /home/sagemuej/local/progwrit/haskell/notmine/haskell/cabal/.stack-work/install/x86_64-linux/ghc-8.0.1/8.0.1/bin/cabal configure -v3
no user package environment file found at /tmp/cbt
Searching for ghc in path.
Found ghc at /usr/local/bin/ghc
/usr/local/bin/ghc --numeric-version
/usr/local/bin/ghc is version 8.3.20170622
looking for tool ghc-pkg near compiler in /usr/local/bin
candidate locations:
["/usr/local/bin/ghc-pkg","/usr/local/bin/ghc-pkg-ghc-8.3.20170622","/usr/local/bin/ghc-pkg-8.3.20170622","/usr/local/bin/ghc-pkg"]
found ghc-pkg in /usr/local/bin/ghc-pkg
/usr/local/bin/ghc-pkg --version
/usr/local/bin/ghc-pkg is version 8.3.20170622
/usr/local/bin/ghc --supported-languages
/usr/local/bin/ghc --info
creating dist
Reading installed packages...
/usr/local/bin/ghc-pkg dump --global -v0
/usr/local/bin/ghc-pkg dump --user -v0
/usr/local/bin/ghc --print-libdir
Reading available packages...
index-state(hackage.haskell.org) = 2017-07-06T13:22:11Z
Searching for pkg-config in path.
Found pkg-config at /usr/bin/pkg-config
/usr/bin/pkg-config --version
/usr/bin/pkg-config is version 0.29.1
/usr/bin/pkg-config --list-all
...
```
Whereas with the GHC-HEAD-built version of cabal-install,
```bash
[21:24:32]sagemuej@sagemuej-X302LA:/tmp/cbt
$ cabal configure -v3
no user package environment file found at /tmp/cbt
Searching for ghc in path.
Found ghc at /usr/local/bin/ghc
/usr/local/bin/ghc --numeric-version
/usr/local/bin/ghc is version 8.3.20170622
looking for tool ghc-pkg near compiler in /usr/local/bin
candidate locations:
["/usr/local/bin/ghc-pkg","/usr/local/bin/ghc-pkg-ghc-8.3.20170622","/usr/local/bin/ghc-pkg-8.3.20170622","/usr/local/bin/ghc-pkg"]
found ghc-pkg in /usr/local/bin/ghc-pkg
/usr/local/bin/ghc-pkg --version
/usr/local/bin/ghc-pkg is version 8.3.20170622
/usr/local/bin/ghc --supported-languages
/usr/local/bin/ghc --info
creating dist
Reading installed packages...
/usr/local/bin/ghc-pkg dump --global -v0
/usr/local/bin/ghc-pkg dump --user -v0
/usr/local/bin/ghc --print-libdir
Reading available packages...
^C  # at this point, memory goes through the roof
```

**Comment by 23Skidoo** [comment]
OK, I think we should now report this to GHC devs. BTW, can you try with GHC 8.2 HEAD?

**Comment by bgamari** [comment]
For the record, I have opened [GHC bug #13930](https://ghc.haskell.org/trac/ghc/ticket/13930) to track this issue. It doesn't appear to affect `ghc-8.3`. Unfortunately I'm having trouble bisecting it due to #4509.

**Comment by 23Skidoo** [comment]
Closing in favour of [GHC bug #13930](https://ghc.haskell.org/trac/ghc/ticket/13930).

**Comment by hvr** [comment]
@bgamari how are you building `cabal`? are recompiling latest lib:Cabal or using  the prebuilt one included in your GHC 8.2 snapshot? I had the impression that #4509 was merely a result compiling a new exe:cabal against an too old lib:Cabal...

**Comment by bgamari** [comment]
I'm using a `cabal-install` and `Cabal` both from the current `v2.0` branch.
